### PR TITLE
Dont ignore subnets from command line args if provided

### PR DESCRIPTION
**Pull request by nimrodshn**
JIRA: https://issues.redhat.com/browse/SDA-3367

cc: @igoihman  @vkareh

### CODE REVIEW

File: cmd/create/cluster/cmd.go [hunk 1]
```diff
@@ -225,12 +225,13 @@ func init() {
...
 	)
 
-	flags.StringArrayVar(
+	flags.StringSliceVar(
 		&args.subnetIDs,
 		"subnet-ids",
 		nil,
 		"The Subnet IDs to use when installing the cluster. "+
 			"SubnetIDs should come in pairs; two per availability zone, one private and one public. "+
+			"Subnets are comma separated, for example: --subnet-ids='subnet-1','subnet-2'."+
```
Comment (vkareh) [inline cmd/create/cluster/cmd.go:234]: Remove the single quotes, they should not be necessary once you fix the `Default` used for pre-populating the `GetMultipleOptions` element: `--subnet-ids=subnet-1,subnet-2`:
```
$ rosa create cluster -i --dry-run --subnet-ids=subnet-050b5a070768d2737,subnet-074572b18f5698f1f
I: Interactive mode enabled.
Any optional fields can be left empty and a default will be selected.
? Cluster name: foo
? Multiple availability zones (optional): No
? AWS region: us-east-1
? OpenShift version (optional): 
? Subnet IDs:  [Use arrows to move, space to select, <right> to all, <left> to none, type to filter, ? for more help]
> [x]  subnet-050b5a070768d2737 (us-east-1f)
  [x]  subnet-074572b18f5698f1f (us-east-1a)
  [ ]  subnet-09a60d10e113314e5 (us-east-1a)
  [ ]  subnet-011158850f4d15a53 (us-east-1a)
  [ ]  subnet-0ba3f0ead1f486510 (us-east-1a)
  [ ]  subnet-0c031ef9162b35f04 (us-east-1a)
  [ ]  subnet-0c7b531f9375931b1 (us-east-1f)
```

**Review by vkareh — CHANGES_REQUESTED** [review]
This needs a few changes in addition to my comments. The `GetMultipleOptions` element does not support defaults, so it's not populated from the command-line. One way of doing that:
```diff
diff --git a/pkg/interactive/interactive.go b/pkg/interactive/interactive.go
index b273e75..af65f2f 100644
--- a/pkg/interactive/interactive.go
+++ b/pkg/interactive/interactive.go
@@ -110,12 +110,12 @@ func parseInt(str string) (num int, err error) {
 func GetMultipleOptions(input Input) ([]string, error) {
        var err error
        res := make([]string, 0)
-       dflt, ok := input.Default.(string)
+       dflt, ok := input.Default.([]string)
        if !ok {
-               dflt = ""
+               dflt = []string{}
        }
        question := input.Question
-       if !input.Required && dflt == "" {
+       if !input.Required && len(dflt) == 0 {
                question = fmt.Sprintf("%s (optional)", question)
        }
        prompt := &survey.MultiSelect{
```
and the corresponding usage:
```diff
diff --git a/cmd/create/cluster/cmd.go b/cmd/create/cluster/cmd.go
index c117544..58d5d7f 100644
--- a/cmd/create/cluster/cmd.go
+++ b/cmd/create/cluster/cmd.go
@@ -417,12 +416,18 @@ func run(cmd *cobra.Command, _ []string) {
                mapSubnetToAZ := make(map[string]string)
                mapAZCreated := make(map[string]bool)
                options := make([]string, len(subnets))
+               optionDefaults := make([]string, len(subnetIDs))
                for i, subnet := range subnets {
                        subnetID := awssdk.StringValue(subnet.SubnetId)
                        availabilityZone := awssdk.StringValue(subnet.AvailabilityZone)
 
                        // Create the options to prompt the user.
                        options[i] = setSubnetOption(subnetID, availabilityZone)
+                       for _, subnetArg := range subnetIDs {
+                               if subnetArg == subnetID {
+                                       optionDefaults = append(optionDefaults, setSubnetOption(subnetID, availabilityZone))
+                               }
+                       }
                        mapSubnetToAZ[subnetID] = availabilityZone
                        mapAZCreated[availabilityZone] = false
                }
@@ -432,6 +437,7 @@ func run(cmd *cobra.Command, _ []string) {
                                Help:     cmd.Flags().Lookup("subnet-ids").Usage,
                                Required: false,
                                Options:  options,
+                               Default:  optionDefaults,
                        })
                        if err != nil {
                                reporter.Errorf("Expected valid subnet IDs: %s", err)
```
This means that the options passed in the command line are now pre-selected.

File: cmd/create/cluster/cmd.go [hunk 2]
```diff
@@ -388,7 +389,9 @@ func run(cmd *cobra.Command, _ []string) {
 		os.Exit(1)
 	}
 
-	useExistingVPC := false
+	subnetIDs := args.subnetIDs
+	useExistingVPC := false || len(subnetIDs) > 0
+	reporter.Debugf("Recieved the following subnetIDs: %v", args.subnetIDs)
 	if interactive.Enabled() {
```
[the remark this answers is not in the document]
Comment (nimrodshn) [reply cmd/create/cluster/cmd.go:395]: But this reverts the "defaults" changes you suggested above - I think it makes sense though.

**Comment by nimrodshn** [comment]
@vkareh Done. ptal

**Pull request merged**